src/main/java/nus/climods/logic/LogicManager.java

```diff
@@ -36,13 +39,21 @@ public LogicManager(Model model, Storage storage) {
     }
 
     @Override
-    public CommandResult execute(String commandText) throws CommandException, ParseException {
+    public CommandResult execute(String commandText) throws CommandException, ParseException, IOException {
         logger.info("[User Command] " + commandText);
 
         CommandResult commandResult;
         Command command = CliModsParser.parseCommand(commandText);
         commandResult = command.execute(model);
 
+        switch (command.getCommandWord()) {
+        case (AddCommand.COMMAND_WORD):
+
+        case (DeleteCommand.COMMAND_WORD):
+            storage.saveUserModuleList(model.getUserModuleList());
+            break;
+        default:
+        }
```

Comment:
Let's shift the trigger to `CommandResult` instead?

Reply:
good catch! Shifted it away

Reply:
What I meant was that we could have a flag in `CommandResult` that keeps track whether we need to a save for `UserModuleList` then `LogicManager` should handle the storage op. `CommandResult` should only keep track of a result of a command, I dont think it will be good if we let it handle IO ops

Reply:
fixed